app/models/folio/service_point.rb

```diff
@@ -36,7 +38,12 @@ def pickup_location?
     private
 
     def library_id
-      @library_id ||= Folio::Types.locations.find_by(primary_service_point_id: id).library_id
+      # Not every service point, e.g .RWC, has an associated location or library
+      location = Folio::Types.locations.find_by(primary_service_point_id: id)
+
+      return if location.nil?
+
+      @library_id ||= location.library_id
     end
```

Comment:
i think doing this will get rid of the memoization, since the `find_by` call will always happen...maybe?
it might work to do:
```ruby
@library_id ||= do
    location = Folio::Types.locations.find_by(primary_service_point_id: id)
    location&.library_id
end
```

Reply:
Used Folio::Types.locations.find_by(primary_service_point_id: id)&.library_id
(Chris also had that comment in slack)